### PR TITLE
Fix for scrollview bug

### PR DESCRIPTION
**Pull request by sarahscott**
Fix for scrollview bug

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>No CHANGELOG added.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please add a description to your PR.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:
      - [] `src/lib/containers/__tests__/home-tests.tsx`

    If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by sarahscott** [comment]
Danger is throwing an error for new home tests even though home is not a new class - this pr is ready for review.

File: src/lib/containers/works_for_you.tsx [hunk 1]
```diff
@@ -40,6 +40,9 @@ interface State {
 }
 
 export class WorksForYou extends React.Component<Props, State> {
+  listView: ListView
+  currentScrollOffset: number
```
Comment (alloy) [inline src/lib/containers/works_for_you.tsx]: These should probably be typed as optional, seeing as they’re not set until the first render.

File: src/lib/containers/works_for_you.tsx [hunk 2]
```diff
@@ -128,6 +131,10 @@ export class WorksForYou extends React.Component<Props, State> {
     })
   }
 
+  componentDidUpdate() {
+    this.listView.scrollTo({ y: this.currentScrollOffset + 1, animated: false })
```
Comment (alloy) [inline src/lib/containers/works_for_you.tsx]: If you don’t scroll at all `this.currentScrollOffset` will be `undefined`, right? Does it not break when doing `undefined + 1`?

Comment (alloy) [inline src/lib/containers/works_for_you.tsx]: Ok it doesn’t break, but it also doesn’t produce a great result. Maybe `this.currentScrollOffset` should be set to `0` in the `constructor`, just to be safe? (Then you also don’t need to mark `currentScrollOffset` as optional.

```
$ node
> undefined + 1
NaN
```

File: src/lib/app_registry.tsx [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
 import * as Relay from "react-relay"
 
 import LoadFailureView from "./components/load_failure_view"
 import Spinner from "./components/spinner"
 import Containers from "./containers/index"
 import Routes from "./relay/routes"
 
-class RootContainer extends React.Component<{}, {}> {
+interface Props extends ViewProperties {
+  trigger1pxScrollHack ?: boolean,
```
Comment (alloy) [inline src/lib/app_registry.tsx]: Nit: no whitespace needed before the question mark.

**Comment by alloy** [comment]
Can you add a CHANGELOG entry?

File: src/lib/containers/works_for_you.tsx (hunk 2 above)
Comment (sarahscott) [inline src/lib/containers/works_for_you.tsx]: It definitely runs, so maybe it's treating it as `0`?

File: src/lib/containers/works_for_you.tsx [hunk 4]
```diff
@@ -132,6 +132,7 @@ export class WorksForYou extends React.Component<Props, State> {
   }
 
   componentDidUpdate() {
+    console.log(`offset: ${this.currentScrollOffset}`)
```
Comment (alloy) [inline src/lib/containers/works_for_you.tsx]: Stray log, can you remove that?

File: src/lib/containers/works_for_you.tsx [hunk 5]
```diff
@@ -40,8 +40,8 @@ interface State {
 }
 
 export class WorksForYou extends React.Component<Props, State> {
-  listView: ListView
-  currentScrollOffset: number
+  listView?: ListView
+  currentScrollOffset?: number = 0
```
Comment (alloy) [inline src/lib/containers/works_for_you.tsx]: Wow, does this actually work?

Comment (sarahscott) [inline src/lib/containers/works_for_you.tsx]: it does!

Comment (alloy) [inline src/lib/containers/works_for_you.tsx]: Dope

**Comment by alloy** [comment]
Superb, let us hope that this does indeed work around the issue and send our thanks to @gre https://github.com/facebook/react-native/issues/1831#issuecomment-288349261 🙌

**Comment by gre** [comment]
aha 😅 I do not take any responsibility over this hack \o/ 😅

File: src/lib/containers/works_for_you.tsx (hunk 2 above)
Comment (gre) [inline src/lib/containers/works_for_you.tsx]: `currentScrollOffset?: number = 0` is set to `0` so it's never undefined right ?

**Comment by orta** [comment]
This was a legitimate build failure, please don't merge red PRs - https://travis-ci.org/artsy/emission/builds/228790776#L238

If Danger is failing and you think it's wrong, fix danger in the PR it's not a lot of work.

**Comment by alloy** [comment]
@orta Indeed you’re right, my bad.

File: src/lib/containers/works_for_you.tsx (hunk 2 above)
Comment (alloy) [inline src/lib/containers/works_for_you.tsx]: Oh yeah, good call 👍

/cc @sarahscott

**Pull request merged**